### PR TITLE
Add cli post processing

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Closes https://github.com/Clinical-Genomics/add-new-tech/issues/81
Add cli command for post-process a PacBio SMRT cell

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/post_process/utils.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Get the correct post-processing service based on the run name.
+    Raises:
+        NameError if the run name is not recognized.
+    """
+    if _is_run_pacbio(run_name):
+        return context.pacbio_post_processing_service
+    raise NameError(f"Could not find a post-processing service for run name: {run_name}")
+
+
+def _is_run_pacbio(run_name: str) -> bool:
```
Comment (diitaz93) [inline cg/cli/post_process/utils.py:20]: with this maybe we can get rid of the name validation in the RunDataGenerator

File: cg/cli/post_process/utils.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import re
+
+from cg.models.cg_config import CGConfig
+from cg.services.post_processing.pacbio.post_processing_service import PacBioPostProcessingService
+
+
+def get_post_processing_service_from_run_name(
```
Comment (diitaz93) [inline cg/cli/post_process/utils.py:7]: This method will have to be expanded when more devices are available to post-process

File: cg/cli/post_process/utils.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/cli/post_process/utils.py:20]: If we do it via pattern matching then we can make a pattern to service map. So we don't need to build up if statements.

File: cg/models/cg_config.py [hunk 3]
```diff
@@ -563,6 +581,31 @@ def mutacc_auto_api(self) -> MutaccAutoAPI:
...
+                file_manager=file_manager,
+                metrics_parser=metrics_parser,
+            )
+            service = PacBioPostProcessingService(
+                run_data_generator=run_data_generator,
+                hk_service=hk_service,
+                store_service=store_service,
+            )
+            self.pacbio_post_processing_service_ = service
+        return service
```
Comment (ChrOertlin) [inline cg/models/cg_config.py:607]: Maybe we can group post processing services all together like.

cg_config.post_processing_services.pacbio / illumina / etc. ?

File: cg/cli/post_process/post_process.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    """Command to post-process a sequencing run from any device.
+
+    run-name is the full name of the sequencing unit of run. For example:
+        PacBio: 'r84202_20240522_133539/1_A01'
+    """
+    post_processing_service: PostProcessingService = get_post_processing_service_from_run_name(
+        context=context, run_name=run_name
+    )
+    post_processing_service.post_process(
+        run_name=run_name, sequencing_dir=context.run_instruments.pacbio.data_dir
```
Comment (ChrOertlin) [inline cg/cli/post_process/post_process.py:33]: this hard codes the pacbio run dir. We need to fetch the proper directory for each service as well here.

File: cg/models/cg_config.py (hunk 3 above)
Comment (diitaz93) [reply cg/models/cg_config.py]: fixed

File: cg/cli/post_process/post_process.py (hunk 4 above)
Comment (diitaz93) [reply cg/cli/post_process/post_process.py]: fixed by adding sequencing dir as parameter to PostProcessingService

File: cg/cli/post_process/utils.py (hunk 1 above)
Comment (diitaz93) [reply cg/cli/post_process/utils.py:20]: maybe we can implement this later when we have more post-processing types?

File: cg/cli/post_process/post_process.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+from cg.cli.utils import CLICK_CONTEXT_SETTINGS
+from cg.models.cg_config import CGConfig
+from cg.services.post_processing.abstract_classes import PostProcessingService
+
+LOG = logging.getLogger(__name__)
+
+
+@click.group(name="post-process", context_settings=CLICK_CONTEXT_SETTINGS)
+def post_process_group():
+    """Add sequencing data to the databases."""
```
Comment (ChrOertlin) [inline cg/cli/post_process/post_process.py:17]:
```suggestion
    """Post-processes sequencing run from the following run instruments: PacBio."""
```
extend when implemented more run instruments

File: cg/cli/post_process/utils.py (hunk 1 above)
Comment (diitaz93) [reply cg/cli/post_process/utils.py]: implemented

**Review by ChrOertlin — APPROVED** [review]
💯

**Pull request merged**